pkg/manager/member/tidb_member_manager.go

```diff
@@ -895,7 +897,13 @@ func getNewTiDBSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap)
 	containers = append(containers, c)
 
 	podSpec := baseTiDBSpec.BuildPodSpec()
-	podSpec.Containers = append(containers, baseTiDBSpec.AdditionalContainers()...)
+
+	var err error
+	podSpec.Containers, err = MergePatchContainers(containers, baseTiDBSpec.AdditionalContainers())
+	if err != nil {
+		return nil, errors2.Wrap(err, "failed to merge containers spec")
```

Comment:
Please remove errors2 and optimize the error message according to the other comments.